torch/distributions/geometric.py

```diff
@@ -75,7 +75,12 @@ def probs(self):
     def sample(self, sample_shape=torch.Size()):
         shape = self._extended_shape(sample_shape)
         with torch.no_grad():
-            u = self.probs.new(shape).uniform_(_finfo(self.probs).tiny, 1)
+            if torch._C._get_tracing_state():
+                # [JIT WORKAROUND] lack of support for .uniform_()
+                u = torch.rand(shape, dtype=self.probs.dtype, device=self.probs.device)
+                u = u.clamp(min=_finfo(self.probs).tiny)
+            else:
+                u = self.probs.new(shape).uniform_(_finfo(self.probs).tiny, 1)
```

Comment:
Those ones aren't equivalent (the non-JIT path is faster), which is why I recommended to leave them. But many other cases like this one can be removed